src/occa/internal/lang/modes/oklForStatement.cpp

```diff
@@ -388,8 +388,11 @@ namespace occa {
             }
           });
 
-        // Index is 1 less than count
-        return maxInnerCount ? maxInnerCount - 1 : 0;
+        // @outer index is 1 less than count
+        if(oklAttr_ == "outer")
```

Comment:
I'm not familiar with this project, but I'd like to ask: are you sure you check `oklAttr_` directly?  Maybe you could use `oklForStatement::isOuterLoop()` (declared a few lines above, [here](https://github.com/libocca/occa/blob/main/src/occa/internal/lang/modes/oklForStatement.cpp#L353-L355)):
```suggestion
        if (isOuterLoop())
```
But as I said I don't know this project well, so I might be wrong.  I do notice `oklAttr_` is passed to this function.

Reply:
Good callout, that is cleaner! Kris and I chatted offline and it looks like he found where the issue is coming from but it might not be the right solution. We might be skipping or including extra loop counts in the `forOklForLoopStatements` method. I'll put some time soon to look more into this method, but thank you both for the inputs!